### PR TITLE
[operators][fir_filter] fix some bug on the python bindings side

### PR DESCRIPTION
**Pull request by MaximilienNaveau**
This PR fixes a few bugs regarding the sot operators and the FIR filters entity declaration.

- One can now import the math_smal_entities.py
- The FIR filter are exported with a correct naming that allow their importation in python.
- Creation of a unit-test that import the python file in src/dynamic_graph
- WARNING API BREAK, I renamed some operators so that the naming is consistent through the operators.

### CODE REVIEW

File: src/matrix/operator-python-module-py.cc [hunk 1]
```diff
@@ -102,6 +102,10 @@ BOOST_PYTHON_MODULE(wrap)
   exposeUnaryOp < HomogeneousMatrixToVector>      ();
   exposeUnaryOp < SkewSymToVector>                ();
   exposeUnaryOp < PoseUThetaToMatrixHomo>         ();
+  exposeUnaryOp < HomogeneousMatrixToVector>      ();
```
Comment (jmirabel) [inline src/matrix/operator-python-module-py.cc:105]: This one is already exposed 3 lines above.

**Review by nim65s — CHANGES_REQUESTED** [review]
This looks good to me, but could you add a simple python unit test for those operators ?

**Review by jmirabel — APPROVED** [review]

File: src/matrix/operator-python-module-py.cc (hunk 1 above)
Comment (MaximilienNaveau) [reply src/matrix/operator-python-module-py.cc:105]: nice catch!

**Review by MaximilienNaveau — COMMENTED** [review]
> This looks good to me, but could you add a simple python unit test for those operators ?

I will see what I can do.

**Comment by MaximilienNaveau** [comment]
> This looks good to me, but could you add a simple python unit test for those operators ?

They are already test in tests/matrix/test_operators.cpp

File: src/matrix/operator-python-module-py.cc (hunk 1 above)
Comment (nim65s) [reply src/matrix/operator-python-module-py.cc:105]: except I can't use it:
```python
from dynamic_graph.sot.core import operator
operator.HomogeneousMatrixToVector()
```
→ `AttributeError: module 'dynamic_graph.sot.core.operator' has no attribute 'HomogeneousMatrixToVector'`

But that's working for others, like `HomoToMatrix()`. I don't see the difference.

Comment (nim65s) [reply src/matrix/operator-python-module-py.cc:105]: Ok, because that is renamed `MatrixHomoToPoseUTheta` in the bindings…

**Comment by MaximilienNaveau** [comment]
I am also adding a unit-test that simply import all python files inside:
https://github.com/stack-of-tasks/sot-core/tree/master/src/dynamic_graph/sot/core
This will hopefully catch a bit of python bindings problems.

File: src/matrix/operator-python-module-py.cc (hunk 1 above)
Comment (MaximilienNaveau) [reply src/matrix/operator-python-module-py.cc:105]: That historic naming, we could just take the chance of renaming this properly here.

**Comment by MaximilienNaveau** [comment]
I did a bunch of changes here.
I add a unit-test on the existing python file for the import (at least).
This would hopefully help if we have a change in the entity naming later on.
@nim65s let me know what you think.

**Comment by MaximilienNaveau** [comment]
PS the import unit-test is failing importing `import dynamic_graph.sot.core.utils.thread_interruptible_loop`

```
=========================================================== test session starts ============================================================
platform linux -- Python 3.6.9, pytest-6.1.1, py-1.9.0, pluggy-0.13.1
rootdir: /home/mnaveau/devel/workspace_ros2/src/sot-core
plugins: ament-xmllint-0.8.1, ament-lint-0.8.1, ament-flake8-0.8.1, ament-copyright-0.8.1, ament-pep257-0.8.1, cov-2.5.1, colcon-core-0.6.1
collected 15 items                                                                                                                         

tests/python/python_imports.py ............F..                                                                                       [100%]

================================================================= FAILURES =================================================================
_____________________________________________ PythonImportTest.test_thread_interruptible_loop ______________________________________________

self = <python_imports.PythonImportTest testMethod=test_thread_interruptible_loop>

    def test_thread_interruptible_loop(self):
        try:
>           import dynamic_graph.sot.core.utils.thread_interruptible_loop

tests/python/python_imports.py:80: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    import threading
    import time
    
>   from dynamic_graph.script_shortcuts import optionalparentheses

../../install/lib/python3.6/site-packages/dynamic_graph/sot/core/utils/thread_interruptible_loop.py:17: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    from __future__ import print_function
    
    # Changing prompt
    import sys
    
    from .entity import Entity
>   from .matlab import matlab
E   ModuleNotFoundError: No module named 'dynamic_graph.matlab'

../../install/lib/python3.6/site-packages/dynamic_graph/script_shortcuts.py:17: ModuleNotFoundError

During handling of the above exception, another exception occurred:

self = <python_imports.PythonImportTest testMethod=test_thread_interruptible_loop>

    def test_thread_interruptible_loop(self):
        try:
            import dynamic_graph.sot.core.utils.thread_interruptible_loop
        except ImportError as ie:
>           self.fail(str(ie))
E           AssertionError: No module named 'dynamic_graph.matlab'

tests/python/python_imports.py:82: AssertionError
========================================================= short test summary info ==========================================================
FAILED tests/python/python_imports.py::PythonImportTest::test_thread_interruptible_loop - AssertionError: No module named 'dynamic_graph....
======================================================= 1 failed, 14 passed in 0.32s =======================================================
```

**Comment by MaximilienNaveau** [comment]
I temporarily disable the unit-tests failing because of the import in `dynamic_graph_python`.

**Comment by jmirabel** [comment]
matlab stuff can be removed. They are outdated.

**Comment by MaximilienNaveau** [comment]
Ok but what about this file? `import dynamic_graph.sot.core.utils.thread_interruptible_loop`
It basically get's an error because it imports: `from dynamic_graph.script_shortcuts import optionalparentheses`
Which in essence is not Matlab stuff.
So I guess another PR in dgp is needed to remove the Matlab stuff but I cannot do it from this PR.
This is why I disabled the uni-test fro now.

Please let me know if you both (@jmirabel and @nim65s) are ok with this PR.

**Comment by jmirabel** [comment]
I have no idea of what `thread_interruptible_loop` is...

**Comment by olivier-stasse** [comment]
> Ok but what about this file? `import dynamic_graph.sot.core.utils.thread_interruptible_loop`
> It basically get's an error because it imports: `from dynamic_graph.script_shortcuts import optionalparentheses`
> Which in essence is not Matlab stuff.
> So I guess another PR in dgp is needed to remove the Matlab stuff but I cannot do it from this PR.
> This is why I disabled the unit-test from now.
> 
> Please let me know if you both (@jmirabel and @nim65s) are ok with this PR.

This is coming from dynamic-graph-python.
It used to be a way to directly test the SoT from a python interpreter.

I am attempting a revival of this part, mostly to have the SoT independent from ROS.

**Comment by olivier-stasse** [comment]
The file is coming from here https://github.com/stack-of-tasks/dynamic-graph-python/blob/master/src/dynamic_graph/script_shortcuts.py

**Comment by olivier-stasse** [comment]
matlab.py was removed by https://github.com/stack-of-tasks/dynamic-graph-python/commit/023344718e7cada832fdbbf94ca8c7f31a1c5c60#diff-edb5d6bd70da37ec2211e94b0e7258efae71b1eb516c1af0a4eac2d785604c26

**Comment by olivier-stasse** [comment]
Just to clarify, I am fine with removing it, but maybe it would be nice to think about a nice example of using the SoT directly from a python interpreter. I am open to any suggestion to have it in a clearer way...

**Review by nim65s — APPROVED** [review]
Thanks ! I opened another issue to track the test that is not passing

**Pull request merged**